### PR TITLE
Basic changes required for perpetual retry on 503

### PR DESCRIPTION
**Pull request by geetduggal**

### CODE REVIEW

File: dxda.go [hunk 1]
```diff
@@ -151,19 +151,27 @@ func makeRequestWithHeadersFail(requestType string, url string, headers map[stri
...
+	// Perpetually retry on 503 (e.g. platform downtime/throttling)
+	for {
+		req, err := retryablehttp.NewRequest(requestType, url, bytes.NewReader(data))
+		check(err)
+		for header, value := range headers {
+			req.Header.Set(header, value)
+		}
+		resp, err := client.Do(req)
+		check(err)
+		defer resp.Body.Close()
```
Comment (orodeh) [inline dxda.go]: I don't know what happens to this defer statement if there is, indeed, a 503. Is it only released at function return? If so, you could have a large accumulation of responses.

Comment (geetduggal) [reply dxda.go]: Good point, let me modify it.

File: dxda.go [hunk 2]
```diff
@@ -151,19 +151,27 @@ func makeRequestWithHeadersFail(requestType string, url string, headers map[stri
...
+		status = resp.Status
+		if status == "503" {
+			time.Sleep(5 * time.Second)
```
Comment (orodeh) [inline dxda.go]: Perhaps you want to take a look at how the dx-toolkit handles the backoff: https://github.com/dnanexus/dx-toolkit/blob/master/src/python/dxpy/__init__.py#L361

Comment (geetduggal) [reply dxda.go]: Thanks!  For the initial releases of dxda, I think a heartbeat check every few seconds should suffice and should handle potentially large downtimes.   I will keep an eye out here and see if an exponential strategy like one used in the toolkit would be better applicable in this case as well.

Comment (geetduggal) [reply dxda.go]: Thanks! I updated the code per your suggestion here and our chat conversation

**Pull request merged**